src/cascadia/TerminalSettingsEditor/MainPage.cpp

```diff
@@ -612,7 +581,11 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         const auto profileViewModel{ _viewModelForProfile(newProfile, _settingsClone) };
         profileViewModel.SetupAppearances(_colorSchemesPageVM.AllColorSchemes());
         const auto navItem{ _CreateProfileNavViewItem(profileViewModel) };
-        SettingsNav().MenuItems().InsertAt(index, navItem);
+
+        if (const auto& navItems{ _menuItemSource })
+        {
+            navItems.InsertAt(index, navItem);
+        }

```

Comment:
Why not just this, like in the other places?
```suggestion
        if (_menuItemSource)
        {
            _menuItemSource.InsertAt(index, navItem);
        }
```

Same question here btw: Can it be null?

Reply:
a bad string replace, that's why